incubator/binding-pgsql/src/main/java/io/aklivity/zilla/runtime/binding/pgsql/internal/stream/PgsqlServerFactory.java

```diff
@@ -0,0 +1,1648 @@
+        PgsqlServer server,
+        long traceId,
+        long authorization,
+        long budgetId,
+        DirectBuffer buffer,
+        int offset,
+        int limit)
+    {
+        final PgsqlSslRequestFW pgsqlSslRequest = sslRequestRO.tryWrap(buffer, offset, limit);
+        final PgsqlCancelRequestMessageFW cancelRequest = cancelReqMessageRO.tryWrap(buffer, offset, limit);
+        final PgsqlStartupMessageFW startupMessage = startupMessageRO.tryWrap(buffer, offset, limit);
+
+        if (pgsqlSslRequest != null &&
+            pgsqlSslRequest.code() == SSL_REQUEST_CODE)
+        {
+            server.decoder = decodePgsqlSslRequest;
+        }
+        else if (cancelRequest != null &&
+                cancelRequest.code() == CANCEL_REQUEST_CODE)
+        {
+            server.decoder = decodePgsqlCancelRequest;
+        }
+        else if (startupMessage != null)
+        {
+            server.decoder = decodePgsqlStartupMessage;
+        }
+        else
+        {
+            server.decoder = decodePgsqlIgnoreAll;
+        }
+
+        return offset;
+    }
+
+    private int decodePgsqlSslRequest(
+        PgsqlServer server,
+        long traceId,
+        long authorization,
+        long budgetId,
+        DirectBuffer buffer,
+        int offset,
+        int limit)
+    {
+        PgsqlSslRequestFW sslRequest = sslRequestRO.wrap(buffer, offset, limit);
+
+        server.onDecodeSslRequest(traceId, authorization);
+        server.decoder = decodePgsqlStartupMessage;
+
+        return sslRequest.limit();
+    }
+
+    private int decodePgsqlStartupMessage(
+        PgsqlServer server,
+        long traceId,
+        long authorization,
+        long budgetId,
+        DirectBuffer buffer,
+        int offset,
+        int limit)
+    {
+        progress.set(offset);
+
+        PgsqlStartupMessageFW startupMessage = startupMessageRO.tryWrap(buffer, offset, limit);
+
+        if (startupMessage != null)
+        {
+            progress.set(startupMessage.limit());
+            Array32FW.Builder<PgsqlParameterFW.Builder, PgsqlParameterFW> parameterBuilder =
+                parametersRW.wrap(messageBuffer, 0, messageBuffer.capacity());
+            final int maxLimit = startupMessage.length() + progress.value - startupMessage.sizeof() - Integer.BYTES;
+            while (progress.value < maxLimit)
+            {
+                final int nameLength = getLengthOfString(buffer, progress.value);
+                final int valueLength = getLengthOfString(buffer, progress.value + nameLength);
+
+                parameterBuilder.item(i -> i
+                    .name(buffer, progress.value, nameLength)
+                    .value(buffer, progress.value + nameLength, valueLength));
+
+                progress.addAndGet(nameLength + valueLength);
+
+                if (buffer.getByte(progress.value) == (byte) 0x00)
+                {
+                    progress.addAndGet(Byte.BYTES);
+                }
```

Comment:
Why is this part conditional?

Reply:
this is an indication of the end of a parameter list